Homework done, added .vs to gitignore file

MegaLexa11 opened the pull request.

MegaLexa11 committed:
Homework done, added .vs to gitignore file

avchirkin commented on Homework/Homework 1.1/Program.cs:
```diff
@@ -0,0 +1,17 @@
+        static void Main(string[] args)
+        {
+            Console.Write("Enter first number: ");
+            int num1 = int.Parse(Console.ReadLine() ?? string.Empty);
```
Если пользователь введёт "мусор", то мы упадём с исключением при попытке парсинга. Посмотри, как работает метод int.TryParse() - давай используем его.

MegaLexa11 replied:
В рамках этого же реквеста отправляю второе ДЗ, с внесенными изменениями в первом

MegaLexa11 replied:
Обнаружил, что забыл про обработку отрицательных значений при вводе, плюс после занятия понял, что там лишнее действие по заполнению массива игроков нулями, раз изначально массив из нулей состоит. Доделаю это

MegaLexa11 committed:
Homework 2 done

В первом дз добавил TryParse вместо Parse. Также отправляю рекурсивное сложение массива с занятия и игру (в ней выполнил вывод цветов немножко не в соответствии с ДЗ)

MegaLexa11 committed:
Added real changes

Добавлена обработка ввода отрицательного значения, исправлена логика создания ловушек и бустов - до этого были ошибки, добавлена обработка последовательного попадания на бусты

MegaLexa11 committed:
Added some more changes

Обработан кейс того, что ловушки могут идти друг за другом, отчего игра может стать непроходимой

MegaLexa11 committed:
Added homework_3 tasks

MegaLexa11 committed:
Added getting page with index

avchirkin commented on Homework/Homework 1.1/Program.cs:
```diff
@@ -0,0 +1,25 @@
+
+            Console.Write("Enter second number: ");
+            bool isNum2 = int.TryParse(Console.ReadLine() ?? string.Empty, out int num2);
+            if (isNum1 && isNum2)
```
Лучше проверять число сразу после ввода. Ввел неправильно первое число - сразу получил ошибку:
`if (!int.TryParse(...)) { Console.WriteLine("Введите число")}`, проверку лучше завернуть в цикл

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+while (isOneMoreTry)
+{
+    Console.Write("Enter number of players: ");
+    bool isPlayersCorrect = int.TryParse(Console.ReadLine() ?? string.Empty, out int playersNum);
```
То же замечание здесь и ниже - выводи ошибку парсинга сразу, не тратя время пользователя

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+
+    if (!(isPlayersCorrect && isSizeCorrect && isTrapsCorrect) || playersNum <= 0 || fieldSize <= 0 || trapsAmount <= 0)
+    {
+        Console.WriteLine("Incorrect Input! You should use only positive numbers");
```
Нене, тут у пользователя слетит крыша - какие числа, в каких настройках? Люби своего пользователя и подсказывай ему более прозрачно

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+        continue;
+    }
+
+    if (fieldSize * trapsPart < trapsAmount)
```
Проверка хорошая, но лучше пользователю подсказать конкретное максимальное число ловушек. Я начинаю играть - мне ещё проценты считать, что ли?)

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+        isCellBlocked = true;
+        while (isCellBlocked)
+        {
+            trapPosition = gameRandomizer.Next(1, fieldSize);
```
Можно сделать проще - например, обходишь карту и генеришь ловушки диапазонами. Первую ловушку получаешь как random.Next(0, N), следующую - random.Next(N + 1, 2N) и так далее. Но на самом деле даже сгенерив N ловушек в диапазоне (0, mapSize) вероятность того, что рандомайзер выдаст 6 клеток с ловушками подряд, околонулевая.

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+        }
+    }
+
+    int boostsAmount = trapsAmount - boostsDifference;
```
Бусты можно расставлять, например, в клетках, где нет ловушек, не завязываясь на фиксированный сдвиг

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+
+    foreach (int pos in trapsPositions) 
+    {
+        Console.Write($"{pos} ");
```
Лучше прокомментировать, что выводишь список клеток с ловушками

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+    Console.WriteLine();
+    foreach (int pos in boostsPositions)
+    {
+        Console.Write($"{pos} ");
```
То же самое - напиши игроку, что выводишь клетки с бустами

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+            Console.ReadLine();
+
+            diceResult = gameRandomizer.Next(1, diceSize);
+            playerPositions[i] += diceResult;
```
Лучше не присваивать позицию сразу, а посчитать, куда попадает игрок, проверить, не является ли клетка ловушкой или бустом, и если игрок попал на ловушку или буст - сообщить об этом и присвоить playerPosition[i] финальное значение сразу.

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+    {
+        for (int i = 0; i < playerPositions.Length; i++)
+        {
+            Console.ForegroundColor = playerColors[i];
```
Форматирование лучше вынести в отдельный метод - PrintCurrentPosition, PrintTrapNotification, PrintBoostNotification и т.п.

avchirkin commented on Homework/Homework2_TheGame/Program.cs:
```diff
@@ -0,0 +1,192 @@
+                    Console.ForegroundColor = playerColors[i];
+                    Console.WriteLine($"{diceResult}.");
+                    playerPositions[i] -= diceResult;
+                    if (playerPositions[i] < 0)
```
playerPosition[i] = Math.Max(playerPosition[i], 0)

avchirkin reviewed (changes requested):
1. Заносить домашки разными ветками в разных pull request
2. Подумать над упрощением реализации логики - сейчас очень много проверок, вложенных друг в друга, логику игры очень непросто читать. Выдели действие "ход" в отдельный метод, внутри которого а) выбрасывается кубик, б) считается новая позиция игрока, в) позиция сравнивается со списком ловушек и бустов, г) исходя из попадания или непопадания в ловушку/буст считается финальная позиция и выводится соответствующее сообщение

MegaLexa11 committed:
Fixed all the problems (probably)

Исправлены замечания (во втором ДЗ, думаю, вышло сделать код более декларативным). Также все сделано в рамках данной ветки, так как пока что структура веток в репозитории совсем сомнительная получилась, хочу это замерджить, а потом уже по каждому дз нормально ветки создавать

MegaLexa11 committed:
Homework 4 done

Calculator program was done

MegaLexa11 committed:
First part of Homework 5 done

Выполнил только первую часть

MegaLexa11 committed:
Merge pull request #2 from MegaLexa11/homework_3

Homework 3

MegaLexa11 committed:
Merge pull request #3 from MegaLexa11/homework_4

Homework 4 done

MegaLexa11 committed:
Removed all unnecessary stuff

Убрал два лишних класса, остатки теперь хранятся полем в рамках Storage, соответствующим образом были адаптированы к этому и другие классы, добавлена цена и ее история. Задачу по свойствам товаров не делал, думаю, что, если и сделаю, то в рамках следующего дз, там она, как будто, поуместнее

MegaLexa11 committed:
Homework_8 done

MegaLexa11 committed:
Some fixes

MegaLexa11 committed:
Merge pull request #4 from MegaLexa11/homework_5

First part of Homework 5 done

MegaLexa11 committed:
Made some fixes

MegaLexa11 committed:
Merge pull request #7 from MegaLexa11/homework_8

Homework_8 done
